### ISSUE TITLE
Assertian failure sequencer_stalls < ktimeout_chunks

### ISSUE THREAD

**Issue by allanjude** [agokhale/viamillipede#1]
```
Assertion failed: (sequencer_stalls < ktimeout_chunks && "transporter accident! rx seqencer stalled"), function rxworker, file rx.c, line 121.
```

**Comment by allanjude** [comment]
One possible cause of this: I was piping the content (a dd image of a hard drive) I was sending through zstd (compressor) before feeding it to viamillipede to deliver it to the remote side. When zstd was finding unallocated space, that was still all 0s, it might not have written anything to the pipe for a long period of time, as it was limited by the read speed of the HDD feeding it zeros which is compressed to nothing. (I think gzip might have still written a few bytes per minute in such a case)

**Comment by agokhale** [comment]
I'm mortified that @allanjude  had to remind me in person that he filed this. It's my belief that agokhale@d5b91d407de44a7dfc9eed417511dbf80012910f  mediates this and will wait for a _long time, thus avoiding transporter accidents  in the heisenburg compensator.

**Comment by keltia** [comment]
Hi, just got the same assertion, I'm using the port version, is this commit in it?

```
centre# viamillipede rx 8834 | zfs receive -F -s tank
Assertion failed: (sequencer_stalls < ktimeout_chunks && "transporter accident! rx seqencer stalled"), function rxworker, file rx.c, line 121.
cannot receive new filesystem stream: checksum mismatch or incomplete stream.
Partially received snapshot is saved.
A resuming stream can be generated on the sending system by running:
    zfs send -t 1-c759db04c-c0-789c636064000310a500c4ec50360710e72765a526973030c49443d460c8a7a515a79680646e2c6281cab321c9275596a41603e980ae9ffb58b0e82fc94f2fcd4c616028cfcc16ceae60d9628024cf0996cf4bcc4d05d28979d9fa6589450e460686968606c6860c080000440a1a0e
zsh: abort (core dumped)  viamillipede rx 8834 |
zsh: exit 1               zfs receive -F -s tank
```

**Comment by keltia** [comment]
Reducing ARC usage fixed my "swap" problem and I can confirm the patch works too.

**Comment by agokhale** [comment]
I'm a bit confused about the out of swap assert; viamillipede should not be allocating memory after the initial setup; was the system. This sequencer assertion fires when the rx deems it's hopelessly slow;  if you might track the current, I've fixed that; I've also made  socket buffer size changes that should greatly increase very high rate connection stability.  I'll tag a  release for ports soon; but if you clone the repo and make install - you might test the new patches.
